### ISSUE TITLE
DELAF-PB lacks 3rd person singular forms of imperfect

### ISSUE THREAD

**Issue by leoalenc** [LFG-PTBR/MorphoBr#8]
Como obervou @odanoburu, o DELAF-PB não tem formas de alguns verbos. Considero isso um erro desse recurso. De fato, os verbos da 2ª conjugação, como "vender", possuem aí tanto as formas da 1ª quanto da 3ª pessoa do singular do imperfeito do indicativo:

grep -E ",vender\.V:I[13]s" Delaf2015v04.dic
vendia,vender.V:I1s
vendia,vender.V:I3s

grep -Ec ",.*er\.V:I1s" Delaf2015v04.dic
800
grep -Ec ",.*er\.V:I3s" Delaf2015v04.dic
800

Quando consideramos a 1ª conjugação, porém, apenas as formas da 1ª pessoa do singular do imperfeito do indicativo estão incluídas:

grep -Ec ",.*ar\.V:I1s" Delaf2015v04.dic
12714
grep -Ec ",.*ar\.V:I3s" Delaf2015v04.dic
0

Não vejo razão para esse tratamento desigual conferido às duas conjugações. Se o argumento para essas lacunas fosse o fato de que uma forma é depreensível da outra, então deveríamos ter também apenas as formas da 1ª pessoa do singular do imperfeito do indicativo da 2ª conjugação. Note que essa identidade de formas é um fenômeno geral do português, que atinge tanto verbos regulares quanto irregulares.
Por outro lado, a filosofia do DELAF-PB parece ser a de outros recursos análogos que se enquadram na categoria dicionário de formas plenas ("full-form lexica"/"full-form dictionaries"). Esse tipo de recurso consiste em listar exaustivamente todas as formas existentes com as suas análises (sob a forma de "word-parse pairs"), essas últimas consistindo de lemas, categoria lexical e demais traços morfossintáticos. Se essa não fosse a filosofia subjacente do DELAF-PB, então esse recurso deveria ter sido consequente nos inúmeros outros casos em que formas são depreensíveis de outras.
Desse modo, proponho que procuremos sistematicamente por outras lacunas desse tipo, calculando que verbos discrepam no número de formas. Os comando abaixo evidenciam que problema análogo existe no mais-que-perfeito:

grep -Ec ",.*ar\.V:Q3s" Delaf2015v04.dic
5
grep -Ec ",.*ar\.V:Q1s" Delaf2015v04.dic
12714

As únicas formas de 3ª pessoa do singular do mais-que-perfeito são estas:

grep -E ",.*ar\.V:Q3s" Delaf2015v04.dic
dera,dar.V:Q3s
desdera,desdar.V:Q3s
estivera,estar.V:Q3s
redera,redar.V:Q3s
sobrestivera,sobrestar.V:Q3s

Vemos que não há lógica por trás da inclusão dessas formas e exclusão das demais.

Depois de feito o levantamento das lacunas, devemos incluir as formas faltantes. Quanto a isso, não há que se preocupar com o tamanho do recurso, uma vez que o tamanho total pode ser reduzido extremamente através da compilação em transdutor de estados finitos (ver, entre outros, Alencar et. al., 2014, p.61)

**Comment by leoalenc** [comment]
Nos comandos acima, usei a barra  inclinada para esquerda para escapar o segundo ponto, antes da categoria V, mas isso sumiu aqui.

**Comment by arademaker** [comment]
@leoalenc acima vc exemplifica lacunas. Nos commits citados, vc resolve as lacunas exemplificadas mas assumo que este issue continuaria aberto dado seu caracter mais amplo certo? 

> Depois de feito o levantamento das lacunas, devemos incluir as formas faltantes. 

Também podemos criar issues específicos para cada lacuna identificada. Vc tem alguma idéia de alguma estratégia para acharmos as lacunas possíveis? Ou trata-se de mesmo uma tarefa manual de recorrer as regras de morfologia e tentar verificar se todas foram contempladas para cada conjugação verbal?

**Comment by arademaker** [comment]
Me ocorreu agora lendo uma resposta do @leoalenc que esta procura de lacunas sistemáticas pode ser feita por contagem de formas para cada tempo verbal e conjugação, certo? É de se esperar uma certa regularidade, correto?

**Comment by leoalenc** [comment]
@odanoburu @arademaker  sim, o número de formas deve ser exatamente o mesmo para cada tempo verbal e cada verbo. Existe o  caso das formas ditas inexistentes dos verbos defectivos, mas acho melhor incluí-las no analisador, Porque os falantes acabam usando essas formas,  que podem ser marcadas com uma etiqueta para evitar a sua geração . Não são muitos os verbos desse tipo em português.

**Comment by leoalenc** [comment]
@odanoburu @arademaker  sim, podemos, a partir de agora, criar uma questão própria pra cada lacuna identificada.

**Comment by odanoburu** [comment]
usando o morphoBR pra gerar as formas GF eu consigo ver as formas ausentes. entre os 282 verbos que analisei, temos essas formas ausentes:
```
grep "nonExist" IrregPor  | sort | uniq -c
     14     {- VPB (Fut Sub Sg P3) => -} "nonExist" ;--#notpresent
      1     {- VPB (Imper Pl P1) => -} "nonExist" ;
      1     {- VPB (Imper Pl P3) => -} "nonExist" ;
    282     {- VPB (Imper Sg P1) => -} "nonExist" ;
      1     {- VPB (Imper Sg P3) => -} "nonExist" ;
    258     {- VPB (MQPerf Sg P3) => -} "nonExist" ;--#notpresent
      1     {- VPB (Pres Ind Sg P1) => -} "nonExist" ;
      1     {- VPB (Pres Sub Pl P1) => -} "nonExist" ;
      1     {- VPB (Pres Sub Pl P2) => -} "nonExist" ;
      1     {- VPB (Pres Sub Pl P3) => -} "nonExist" ;
      1     {- VPB (Pres Sub Sg P2) => -} "nonExist" ;
    258     {- VPB (PretI Ind Sg P3) => -} "nonExist" ;--#notpresent
      1     {- VPB (PretP Sg P1) => -} "nonExist" ;--#notpresent
      1     {- VPB (PretP Sg P3) => -} "nonExist" ;--#notpresent
```

**Comment by arademaker** [comment]
Hum referência sobre a diferença entre formas que não existem mesmo e 

> Existe o caso das formas ditas inexistentes dos verbos defectivos

?

**Comment by odanoburu** [comment]
referência não tenho, mas pelo que entendi do @leoalenc é possível que as pessoas usem formas que supostamente não existem de verbos defectivos, mas não que elas usem a primeira pessoa do imperativo (que no fundo seria o único caso de forma "que não existe mesmo")

**Comment by leoalenc** [comment]
@odanoburu @arademaker No dicionário de São Carlos, o somatório de formas verbais sem pronomes deve ser 71  para cada verbo. Façamos as contas. Cada tempo do indicativo e do subjuntivo deve ter seis formas, além disso o infinitivo flexionado também deve ter seis formas. São seis tempos do indicativo, três do subjuntivo. Com o infinitivo flexionado, são, portanto, 60 formas. O imperativo tem cinco formas, porque não existe a primeira pessoa do singular. O particípio passivo tem quatro formas. O gerúndio e o infinitivo não flexionado possuem, cada um, uma forma.
Segundo a nova gramática do português contemporâneo de Cunha e Cintra, editora nova fronteira, são exemplos de verbos defectivos banir, abolir e falir. Esses verbos não possuiriam o subjuntivo presente. Um verbo do tipo banir não possuiria a primeira pessoa do singular do presente do indicativo. Já verbos do tipo de falir só possuiriam no presente do indicativo a primeira pessoa do plural e a segunda pessoa do plural.  No imperativo, os verbos defectivos também não possuiriam todas as formas, dependendo do grupo do verbo.
No dicionário de São Carlos, todos esses três verbos classificados como defectivos apresentam 71 formas, portanto, são conjugados de forma completa.  Eu não tenho nenhum problema com isso, dada a minha orientação linguística gerativa. A linguística gerativa distingue entre conhecimento linguístico, ou seja, a competência, e a performance ou desempenho ou comportamento linguístico ou uso da língua em situações concretas. Na minha opinião, sob a perspectiva gerativa, a gramática morfológica dos falantes inclui as formas consideradas defectivas como formas potenciais, tanto é que todo falante é capaz de produzir essas formas! As chamadas formas defectivas não são usadas ou não devem ser usadas conforme os gramáticos tradicionais por uma série de motivos, um deles é a cacofonia. Ora, muitos falantes não são sensíveis a essa questão da cacofonia, podendo vir a produzir as formas proibidas pelos gramáticos tradicionais. Outros verbos defectivos se referem as vozes dos animais, como, por exemplo, latir. Como cães não falam, formas como latimos não deveriam existir conforme a gramática tradicional. A linguística gerativa, porém, não padece dessa limitação de imaginação, porque autores podem muito bem descrever situações em que cães falam ou usar isso de forma metafórica para se referir a pessoas. Esse exemplo mostra que devemos ter sempre uma grande desconfiança em relação à gramática tradicional, assim como os físicos devem colocar em suspeição as ideias dos antigos gregos sobre a natureza. Em grande parte, a gramática tradicional apenas repete coisas sem fundamento científico que se afirmam desse a antiguidade clássica. Concluindo, minha recomendação é que nós deixamos as conjugações completas dos chamados de verbos defectivos. Um léxico computacional deve ser um modelo do léxico mental dos falantes, onde essas formas chamadas defectivas existem potencialmente, assim como os diminutivos de todos os substantivos e adjetivos, assunto de que  já tratei em comentário a outra questão.

**Comment by leoalenc** [comment]
@odanoburu e @arademaker, para mim, deixar os chamados verbos  defectivos  com todas as suas 71 formas, como está no dicionário de São Carlos, seria ponto pacífico.  Espero que vocês também não tenho problemas com isso. Por outro lado, o que realmente deve causar preocupação são as lacunas na conjugação de muitos  verbos. Eu já preenchi lacunas para o caso dos verbos da primeira conjugação no pretérito imperfeito e no mais que perfeito, conforme  questão específica levantada. Agora descobri, por acaso, que o verbo vender não possui a terceira pessoa do singular do futuro do subjuntivo!

**Comment by leoalenc** [comment]
@odanoburu @arademaker O verbo sofrer também tem a mesma lacuna que vender!  Não há nenhuma razão para isso, é bug mesmo!

**Comment by odanoburu** [comment]
>  Concluindo, minha recomendação é que nós deixamos as conjugações completas dos chamados de verbos defectivos. 

concordo, @leoalenc !

> Espero que vocês também não tenho problemas com isso.

nenhum!

**Comment by arademaker** [comment]
@odanoburu no seu comentário https://github.com/LFG-PTBR/MorphoBr/issues/8#issuecomment-381406441 vc poderia explicar a saída? Por que algumas linhas tem `--#notpresent` e outras não? Vc consegue identificaf os verbos que tem estas lacunas?

**Comment by odanoburu** [comment]
@arademaker eu fiz o grep num módulo GF, o `--#notpresent` é um marcador dos tempos que não fazem parte dos tempos presente (é assim que GF diferencia os imports `import present/LangPor.gfo` e `import alltenses/LangPor.gfo`).

> Vc consegue identificaf os verbos que tem estas lacunas?

não tão facilmente como temos feito usando grep, pq as funções não são line-oriented como o dicionário..

**Comment by arademaker** [comment]
Relacionado com #22
